src/Exception/RequestException.php

```diff
@@ -142,6 +146,24 @@ public static function getResponseBodySummary(ResponseInterface $response)
     }
 
     /**
+     * Obfuscates URI if there is an username and a password present
+     *
+     * @param UriInterface $uri
+     *
+     * @return UriInterface
+     */
+    private static function obfuscateUri($uri)
```

Comment:
the call of this function happens earlier than we have exception object. So, it has to be a static function.